### DISCUSSION TITLE
Tuya TRV601Z / TRV602Z - thermostatic radiator valve - update converter

### DISCUSSION THREAD

**Discussion by spiderjoka** [Koenkk/zigbee2mqtt#25269] [Devices]
Hi

I recently bought some of this devices, but it showed up that some features still not working, like Auto Mode and Home Assistant climate entity e.g..

https://www.zigbee2mqtt.io/devices/TRV602Z.html

I changed the actual converter an made a custom one and just want to share it here. Auto Mode is working and HA climate entity works as best as possible. Its not best solution and to make auto mode workin out of the box, it needs some more attention and different code, but have no time for it.

Its for the device  TRV601Z and 602Z (same devices just different display) and maybe some other similar devices, but with different datapoints and factorycode.

[trv601z_trv602z.txt](https://github.com/user-attachments/files/18200696/trv601z_trv602z.txt)

**Comment by waclaw66** [comment]
I've tested your config for _TZE204_ltwbm23f and it works perfectly (including scheduled temps), could you or @Koenkk please make a PR to fix the device in main? Thanks
This converter also fixes https://github.com/Koenkk/zigbee2mqtt/issues/24795

**Reply by Koenkk** [reply]
You can make a PR by clicking [here](https://github.com/Koenkk/zigbee-herdsman-converters/edit/master/src/devices/tuya.ts)

**Reply by g-chevalier** [reply]
Thanks @spiderjoka for this work!

I confirm schedulings are working with this converter.
But it's a work in progress and I think has to be handled by the dev team for a perfect integration into the Z2M code.
For information, I created an issue [mentioning this solution](https://github.com/Koenkk/zigbee2mqtt/issues/26195).

**Comment by andreypuhovsky** [comment]
hi @spiderjoka @waclaw66 here you go https://github.com/Koenkk/zigbee-herdsman-converters/issues/8819

**Reply by waclaw66** [reply]
Thanks, however current implementation is fully functional for me.
